src/server/bg_services/tier2_block_evicter.js

```diff
@@ -0,0 +1,134 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const _ = require('lodash');
+
+const P = require('../../util/promise');
+const dbg = require('../../util/debug_module')(__filename);
+const config = require('../../../config');
+const MDStore = require('../object_services/md_store').MDStore;
+const system_store = require('../system_services/system_store').get_instance();
+const system_utils = require('../utils/system_utils');
+const map_server = require('../object_services/map_server');
+
+class Tier2BlockEvicter {
+    /**
+     * Tier2BlockEvicter constructor
+     * @param {{ name: any; client: nb.APIClient }} params 
+     */
+    constructor({ name, client }) {
+        this.name = name;
+        this.client = client;
+    }
+
+    run_batch() {
+        if (!system_store.is_finished_initial_load) {
+            dbg.log0('TIER2_BLOCK_EVICTER: system_store did not finish initial load');
+            return;
+        }
+
+        const system = system_store.data.systems[0];
+        if (!system || system_utils.system_in_maintenance(system._id)) return;
+
+        return this.run_block_evicter();
+    }
+
+    async run_block_evicter() {
+        if (!this.marker) {
+            dbg.log0('TIER2_BLOCK_EVICTER:', 'BEGIN');
+        }
+
+        try {
+            // TODO: Improve the iteration here - Instead of iterating
+            // over all the blocks, we should iterate over the blocks
+            // that have `node_type` set to `BLOCK_STORE_FS`.
+            const blocks = await this.iterate_all_blocks(
+                this.marker,
+                config.AGENT_BLOCKS_RECLAIMER_BATCH_SIZE,
+                false /* deleted_only */
+            );
```

Comment:
The question is if we need to add time conditions to the evictor, so that it won't immediately evict blocks that were just read and premigrated. We might need to set a configurable time before it should get remigrated to avoid thrashing.

Reply:
@guymguym how about something like this?

```diff
async populate_block_evicter_blocks(blocks) {
    if (!blocks || !blocks.length) return;

    const db_blocks = await this.populate_nodes_for_blocks(blocks);

    // Filter out the blocks which have `node_type` set to `BLOCK_STORE_FS`.
-   return _.filter(db_blocks, block => block.to_block_md().node_type === 'BLOCK_STORE_FS');
+   return _.filter(db_blocks, block => {
+       const is_blocks_fs = block.to_block_md().node_type === 'BLOCK_STORE_FS';
+       if (!is_blocks_fs) return false;
+
+       // @ts-ignore
+       const last_read = block.chunk.chunk_db.tier_lru;
+       if (last_read instanceof Date) {
+           const curr_time = moment();
+           if (curr_time.diff(last_read, 'minutes') < config.TIER2_BLOCK_EVICTER_MIN_LRU_TIME) {
+               return false;
+           }
+       }
+ 
+       return true;
+     });
}
```